### ISSUE TITLE
crash app on ipad When you take a picture and then take another picture in xamarin forms

### ISSUE THREAD

**Issue by musab-alrmmah** [jamesmontemagno/MediaPlugin#696]
crash app with without error message  on ipad When you take a picture and then take another picture in xamarin forms

**Comment by jamesmontemagno** [comment]
Would need a repro or crash log. I have taken hundreds of photos back to back.

**Comment by neil-civica** [comment]
I have seen the exact same issue on iPad, the app freezes / crashes on iOS version 12.3.1 running on iPad ONLY (it works on 12.3.1 on an iPhone).

Looking at the iOS logs it suggests a timeout where the camera is not launched a certain time.

It is worth pointing out it does not happen every time though, if you need any more info please let me know

What happens is when it tries to launch the camera the camera does not appear so the app is left in the foreground, when you try to interact with the app like pressing software back button or any other control nothing happens as it is frozen so you have to kill it

**Comment by musab-alrmmah** [comment]
@jamesmontemagno  No error message  or crash log but it freezes and then stops

**Comment by josesimas** [comment]
I just got the same exact same bug report and I can duplicate it. I can take as many photos as I want on an IPhone (tried 5S, 6S and XS) but when I take two photos on an iPad (2018) or on an iPad Pro (2018) the call to TakePhotoAsync does not return (when I take the second picture) and after 2 or 3 minutes the app crashes. I have AppCenter configured but I don't get any exception report for this crash. When I debug using VS2019 I also don't get an exception, the app just unloads after that period of time.

@jamesmontemagno would you try this on an iPad 2018 running the latest iOS and let us know if it worked for you?

I got a device log for both the iPhone and the iPad and the latter has these erros:

 default	11:14:19.954574 +0100	IrisHH.iOS	Call active status: Inactive (initial query)
 default	11:14:19.962109 +0100	IrisHH.iOS	Capture orientation LandscapeLeft (initial)
 default	11:14:19.965604 +0100	IrisHH.iOS	Photo/BackAuto (requestID=0): Calling -[AVCaptureSession beginConfiguration]
 default	11:14:19.967101 +0100	IrisHH.iOS	Paused iris video capture
 default	11:14:19.967920 +0100	IrisHH.iOS	Photo/BackAuto (requestID=0): Calling -[AVCaptureSession commitConfiguration]
 default	11:14:19.968538 +0100	IrisHH.iOS	Photo/BackAuto (requestID=0): Finished -[AVCaptureSession commitConfiguration] (took 0.000 seconds)
 default	11:14:19.968641 +0100	IrisHH.iOS	CAMCaptureEngine start (id=3: "initial configuration"): Session is not running. Calling -[AVCaptureSession startRunning].
 error	11:14:19.974277 +0100	IrisHH.iOS	CGContextAddPath: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
 error	11:14:19.974377 +0100	IrisHH.iOS	clip: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
 error	11:14:19.974481 +0100	IrisHH.iOS	CGContextSetFillColorWithColor: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
 error	11:14:19.974581 +0100	IrisHH.iOS	CGContextFillRects: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
 error	11:14:19.974681 +0100	IrisHH.iOS	CGContextAddPath: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
 error	11:14:19.974788 +0100	IrisHH.iOS	clip: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
 error	11:14:19.974879 +0100	IrisHH.iOS	CGContextDrawLinearGradient: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
 ...
 error	11:16:45.344025 +0100	assertiond	[IrisHH.iOS:2061] SyscallError: setpriority(PRIO_DARWIN_ROLE, 2061, 3): No such process

I cannot find a way to attach the full logs to this post but I can send them to you if it helps.

**Comment by josesimas** [comment]
Just got a message from a customer that picking an image twice also crashes the app. I cannot duplicate it on the iPad Pro but the user has an iPad. This is a pity as I am left without a workaround.

**Comment by musab-alrmmah** [comment]
Yes the problem still exists. Is there a solution?

**Comment by josesimas** [comment]
I am stuck, I looked for an alternative way to take/pick pictures in Xamarin iOS but everyone refers to this plugin.

**Comment by mhBahrami** [comment]
@jamesmontemagno I have the same issue of freezing the iOS app after taking a picture by camera. Any suggestion?

**Comment by yalamandarao** [comment]
Yes I'm also facing same issue iOS 12.4.3 app crash.. but no error message

**Comment by Alex6511** [comment]
Did anyone find a solution to this?

**Comment by Alex6511** [comment]
@jamesmontemagno For repo, I noticed it happened on an iPhone 6 plus and it seems to happen more frequently if you attempt to zoom while taking the 2nd picture.

**Comment by Alex6511** [comment]
After thinking about it more, is it possible this is related to using a slider like #450 @musab-alrmmah @josesimas Do either of you use sliders in your app?

**Comment by josesimas** [comment]
@Alex6511 no, no sliders at all in my app.

**Comment by josesimas** [comment]
But I do use Grailkit @Alex6511 !

**Comment by Alex6511** [comment]
@josesimas I think this is being caused by Grailkit overriding the appearance of the zoom slider, but I have no idea how to fix it.

**Comment by Alex6511** [comment]
@josesimas Possible solution, check this out https://github.com/UXDivers/Grial-UI-Kit-Support/issues/333

**Comment by josesimas** [comment]
@Alex6511 many thanks, I will have a go as soon as possible.

**Comment by Alex6511** [comment]
Doing the above solution resolved this issue for me. @musab-alrmmah if you use grailkit try the solution in the github above.
